src/components/SnippetCard.tsx

```diff
@@ -0,0 +1,125 @@
+import type React from 'react'
+import { useState } from 'react'
+import { Share2 } from 'lucide-react'
+
+import Upvote from 'assets/upvote.svg'
+import Upvoted from 'assets/upvoted.svg'
+import UpvoteHover from 'assets/upvote_hover.svg'
+import Star from 'assets/star.svg'
+import Starred from 'assets/starred.svg'
+import StarHover from 'assets/star_hover.svg'
+import { Button } from 'components/ui/button'
+import LiveblocksComments from 'components/LiveblocksComments'
+
+interface Snippet {
+  id: number;
+  radio_code: string;
+  name: string;
+  type: string;
+  channel: string;
+  state: string;
+  human_upvotes: number;
+}
+
+interface SnippetCardProps {
+  snippet: Snippet;
+  onSnippetClick: (id: number) => void;
+}
+
+const SnippetCard: React.FC<SnippetCardProps> = ({ snippet, onSnippetClick }) => {
+  const [isUpvoted, setIsUpvoted] = useState(false)
+  const [isHovered, setIsHovered] = useState(false)
+  const [isStarred, setIsStarred] = useState(false)
+  const [isStarHovered, setIsStarHovered] = useState(false)
+
+  const handleUpvoteClick = (e: React.MouseEvent) => {
+    e.stopPropagation()
+    setIsUpvoted(!isUpvoted)
+  }
+
+  const getUpvoteButtonClasses = () => {
+    const baseClasses = 'font-normal rounded-full border-none'
+    return isUpvoted
+      ? `${baseClasses} bg-gradient-to-r from-blue-deep to-blue-rich text-white hover:from-blue-deep hover:to-blue-rich hover:text-white`
+      : `${baseClasses} bg-blue-light text-blue-accent hover:bg-blue-light hover:text-blue-accent`
+  }
+
+  const handleStarClick = (e: React.MouseEvent) => {
+    e.stopPropagation()
+    setIsStarred(!isStarred)
+  }
```

Comment:
--- $${\color{green}[AI \space Generated \space Comment]}$$ ---
The `handleUpvoteClick` and `handleStarClick` functions update the local state but do not persist these actions to a backend service. Consider integrating API calls within these handlers to update the backend accordingly:

```typescript
const handleUpvoteClick = async (e: React.MouseEvent) => {
  e.stopPropagation()
  try {
    await updateUpvoteOnBackend(snippet.id, !isUpvoted)
    setIsUpvoted(!isUpvoted)
  } catch (error) {
    console.error('Failed to update upvote:', error)
  }
}
```